core/src/test/java/com/backblaze/b2/client/B2ListFileNamesIterableTest.java

```diff
@@ -69,6 +71,49 @@ public void testOnePage() throws B2Exception {
         assertTrue(!iter.hasNext());
     }
 
+    @Test
+    public void testEmptyPageWithNextFileName() throws B2Exception {
+          final B2ListFileNamesRequest request = B2ListFileNamesRequest
+                .builder(BUCKET_ID)
+                .setStartFileName("file/0000")
+                .setMaxFileCount(106)
+                .setPrefix("file/")
+                .setDelimiter("/")
+                .build();
+
```

Comment:
does it need a test where the first page has results, but the second is empty and has a next file name? (not sure if that would go through a different code path)